reframe/utility/lmod_helper.py

```diff
@@ -0,0 +1,67 @@
+import subprocess
+
+class PopenCommunicate():
+    def exec(self, command):
+        proc=subprocess.Popen(command,
+                              shell = True,
+                              stderr=subprocess.STDOUT,
+                              stdout=subprocess.PIPE)
+        return str(proc.communicate()[0]) # None-terminated, removed None
+
+
+def _all_avail_mods(thing_to_load, ml_av_output):
+    all_versions = []
+    look_for =  thing_to_load + "/"
+    for item in ml_av_output.replace("\n", " ").split(" "):
+        if item.strip().startswith(look_for):
+            all_versions.append(item.strip().split("/")[1])
+
+    return all_versions
+
+def _loop_among_one(module_glob, pc, env_context=""):
```

Comment:
I also think this should go to the Lmod backend. 
The python bindings for Lmod are preferred over calling `ml avail`.
And the function name should be slightly improved to better explain what it does.